gateway/ln-gateway/src/lib.rs

```diff
@@ -253,15 +292,13 @@ impl Gateway {
     }
 
     async fn handle_get_info(&self, _payload: InfoPayload) -> Result<GatewayInfo> {
-        let federations = self
-            .actors
-            .lock()
-            .await
-            .iter()
-            .map(|(_, actor)| actor.get_info().expect("Failed to get actor info"))
-            .collect();
+        let actors = self.actors.lock().await;
+        let mut federations: Vec<FederationInfo> = Vec::new();
+        for (_, actor) in actors.iter() {
+            federations.push(actor.read().await.get_info()?);
+        }
 
-        let ln_info = self.lnrpc.info().await?;
+        let ln_info = self.lnrpc.read().await.info().await?;
```

Comment:
Ideally we should check if the lightning node is connected or not. And return a value indicating if its connected or not.

Reply:
Actually I was mistaken on this, this is already taken care of. `info` will check if we're connected and return an error if its not connected.